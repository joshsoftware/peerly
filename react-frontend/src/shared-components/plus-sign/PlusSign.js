import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const PlusSignWrapper = styled.div`
  transform: ${({ cross }) => (cross ? "rotate(45deg)" : "rotate(0deg)")};
  font-size: ${({ fontSize }) => fontSize};
  color: ${({ theme }) =>
    theme === "dark" ? "var(--white)" : "var(--atomic)"};
`;

const PlusSign = ({ theme, cross, fontSize }) => (
  <PlusSignWrapper
    theme={theme}
    cross={cross}
    fontSize={fontSize}
    data-testid="plusSignText"
  >
    +
  </PlusSignWrapper>
);

PlusSign.defaultProps = {
  theme: "dark",
  fontSize: "36px",
  cross: false,
};

PlusSign.propTypes = {
  theme: PropTypes.oneOf(["dark", "light"]),
  cross: PropTypes.bool,
  fontSize: PropTypes.string,
};

export default React.memo(PlusSign);
