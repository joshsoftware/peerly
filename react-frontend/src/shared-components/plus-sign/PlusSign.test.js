import React from "react";
import { render } from "@testing-library/react";

import PlusSign from "shared-components/plus-sign/PlusSign";

it("should equal to + text", () => {
  const { getByTestId } = render(<PlusSign />);
  expect(getByTestId("peerlyplussign")).toHaveTextContent("+");
});
