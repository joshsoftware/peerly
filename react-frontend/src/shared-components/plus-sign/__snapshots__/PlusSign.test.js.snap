// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render plus sign component 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  font-size: 48px;
  color: var(--white);
}

<div
    class="c0"
    data-testid="plusSignText"
    font-size="48px"
  >
    +
  </div>
</DocumentFragment>
`;
